### PR TITLE
docs: replace `dotenv` with `--env-file`

### PR DESCRIPTION
**Pull request by hyunbinseo**
- Node.js v20 LTS started in v20.9.0: [release post](https://nodejs.org/en/blog/release/v20.9.0)
- The `--env-file` flag was added in Node.js v20.6.0: [release post](https://nodejs.org/en/blog/release/v20.6.0)
- It is the advised way of importing environment variables: [official guide](https://nodejs.org/en/learn/command-line/how-to-read-environment-variables-from-nodejs)

---

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [ ] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [ ] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [ ] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

### Edits

- [x] Please ensure that 'Allow edits from maintainers' is checked. PRs without this option may be closed.

### CODE REVIEW

**Comment by Conduitry** [comment]
SvelteKit only requires Node 18.13+ so it seems to me like a bad idea to completely get rid of the `dotenv` recommendation in favor of something that only works in 20.6+.

**Comment by hyunbinseo** [comment]
@Conduitry Should this be merged in Node.js v18 EoL 2025-04-30? Or should we wait for SvelteKit to support v20+?

**Comment by PuruVJ** [comment]
Ideally shouldn't do this before Kit's new major drops support for v18

**Comment by dominikg** [comment]
This is only a documentation change highlighting the new api for users of node20.6+, right?

Instead of mixing the two approaches i would keep the old 18+ documentation as is, and add a new info-box or paragraph 

like 

> If you use node20.6+ you can also use the new --env flag instead of dotenv ... 
> ```bash 
>  ... example here ...
> ```

File: documentation/docs/25-build-and-deploy/40-adapter-node.md [hunk 1]
```diff
@@ -51,8 +51,15 @@ npm install dotenv
 ...and invoke it before running the built app:
 
 ```diff
--node build
-+node -r dotenv/config build
+- node build
++ node -r dotenv/config build
```
Comment (dominikg) [inline documentation/docs/25-build-and-deploy/40-adapter-node.md:55]: is this change needed? the text above says to invoke dotenv before running the *built*  app, but this is during build.
Sveltekit uses vite which has dotenv builtin https://vitejs.dev/guide/env-and-mode

File: documentation/docs/25-build-and-deploy/40-adapter-node.md [hunk 2]
```diff
@@ -51,8 +51,15 @@ npm install dotenv
...
+```
+
+If you use Node.js v20.6+, you can use the [`--env-file`](https://nodejs.org/en/learn/command-line/how-to-read-environment-variables-from-nodejs) flag instead:
+
+```diff
+- node build
++ node --env-file=.env --env-file=.env.production build
```
Comment (dominikg) [inline documentation/docs/25-build-and-deploy/40-adapter-node.md:62]: same as above really. instead of showing how to pass this on build, it would have to be shown when starting the app :thinking:

File: documentation/docs/25-build-and-deploy/40-adapter-node.md (hunk 1 above)
Comment (hyunbinseo) [reply documentation/docs/25-build-and-deploy/40-adapter-node.md:55]: The default build directory is `./build`.

`node build` is how you start the `adapter-node` application.

https://kit.svelte.dev/docs/adapter-node#deploying

```
📦build
 ┣ 📂client
 ┣ 📂prerendered
 ┣ 📂server
 ┣ 📜env.js
 ┣ 📜handler.js
 ┣ 📜index.js
 ┗ 📜shims.js
```

Comment (dominikg) [reply documentation/docs/25-build-and-deploy/40-adapter-node.md:55]: oh right :see_no_evil:

**Review by dominikg — APPROVED** [review]

File: documentation/docs/25-build-and-deploy/40-adapter-node.md (hunk 2 above)
Comment (benmccann) [inline documentation/docs/25-build-and-deploy/40-adapter-node.md:62]: I suspect it's uncommon to have two env files. It makes the command look rather long to show it that way by default. I think if we wanted to show both it might be better to do it the way the node docs do where it shows it with a single one first and then shows a second example with multiple flags
```suggestion
+ node --env-file=.env build
```

Comment (hyunbinseo) [reply documentation/docs/25-build-and-deploy/40-adapter-node.md:62]: Does SvelteKit bundle all `.env.*` files supported by Vite?

I personally have a separate `.env.production` file for the Node.js production server.

Which is why I've added `--env-file=.env.production` in the documentation code.

```dotenv
# (adapter-node) -----------------------------------------------------------------------------------

# Reference https://kit.svelte.dev/docs/adapter-node#environment-variables

# The URL of the deployment should be provided for the CSRF protection (check origin) to work.
# If not, the server will respond with a [403] Cross-site POST form submissions are forbidden.

ORIGIN="" # https://your-domain.here

# `CF-Connecting-IP` provides the client IP address connecting to Cloudflare to the origin web server.
# Reference https://developers.cloudflare.com/fundamentals/reference/http-request-headers/

ADDRESS_HEADER="CF-Connecting-IP"
```

As suggested, `--env-file=.env` would be much simple, and match the dotenv example.

```shell
# Dotenv does support custom config path
node -r dotenv/config your_script.js dotenv_config_path=/custom/path/to/.env dotenv_config_debug=true
```

Comment (benmccann) [reply documentation/docs/25-build-and-deploy/40-adapter-node.md:62]: The way I'd expect it to be used is:
- there is no `.env` file checked in except for something like `.env.sample`, `.env.local`/`.env.development`, `.env.test`, etc.
- the app is built on a CI. The CI provides the `.env` file or environment variables

Comment (hyunbinseo) [reply documentation/docs/25-build-and-deploy/40-adapter-node.md]: Never considered using CI with the Node.js adapter.

Since the Node.js docs is linked, removing duplicate flags seems reasonable.
